File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/service/UserService.java
package com.pds.localpos.userservice.service;

import com.pds.localpos.userservice.dto.request.UserRequest;
import com.pds.localpos.userservice.dto.response.UserResponse;
import com.pds.localpos.userservice.model.User;

import java.util.List;

public interface UserService {

    UserResponse createUser(UserRequest dto);

    UserResponse getUserById(String id);

    List<UserResponse> getAllUsers();

    UserResponse updateUser(String id, UserRequest dto);

    void deleteUser(String id);

    User findByUsername(String username);
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/service/UserService.kt
package com.pds.localpos.userservice.service

import com.pds.localpos.userservice.dto.UserRequestDTO
import com.pds.localpos.userservice.dto.UserResponseDTO
import com.pds.localpos.userservice.model.User

interface UserService {

    fun createUser(dto: UserRequestDTO): UserResponseDTO

    fun getUserById(id: String): UserResponseDTO

    fun getAllUsers(): List<UserResponseDTO>

    fun updateUser(id: String, dto: UserRequestDTO): UserResponseDTO

    fun deleteUser(id: String)

    fun findByUsername(username: String): User?
}
